src/computability/DFA.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+-/
+
+import data.fintype.basic
+
+/-!
+# Deterministic Finite Automata
+This file contains the definition of a Deterministic Finite Automaton (DFA), a state machine which
+determines whether a string (implemented as a list over an arbitrary alphabet) is in a regular set
+in linear time.
+-/
+
+universes u v
+
+/-- A DFA is a set of states (`state`), a transition function from state to state labelled by the
+  alphabet (`step`), a starting state (`start`) and a set of acceptance states (`accept`). -/
+structure DFA (alphabet : Type u) (state : Type v) :=
```

Comment:
```suggestion
structure DFA (α : Type u) (σ : Type v) :=
```
You're using `α` and `σ` in the rest of the file.  I think it's better to be consistent than descriptive here.  (Also `α` and `σ` are good mnemonics anyhow!)